### PR TITLE
Use case apt application

### PR DESCRIPTION
**Pull request by nitro-neal**
This is a full example flow of an apartment verifying the age of a potential tenant.

### CODE REVIEW

File: example/use_cases/apartment_application/apartment_application.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+)
+
+func main() {
+
+	/**
+		 Step 1: Create new entities as DIDs. Govt Issuer, User Holder, and Apartment Verifier
+	**/
+
+	// User Holder
+	holderDIDPrivateKey, holderDIDKey, _ := did.GenerateDIDKey(crypto.Ed25519)
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:35]: the other examples do some light error handling like this, what do you think? https://github.com/TBD54566975/ssi-sdk/blob/main/example/vc/vc.go#L54

File: example/use_cases/apartment_application/apartment_application.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	holderVerifier, _ := cryptosuite.NewJSONWebKeyVerifier(holderJWK.ID, holderJWK.PublicKeyJWK)
+
+	// Apt Verifier
+	aptDidPrivateKey, aptDIDKey, _ := did.GenerateDIDKey(crypto.Ed25519)
+	aptJWK, _ := cryptosuite.JSONWebKey2020FromEd25519(aptDidPrivateKey.(ed25519.PrivateKey))
+	aptSigner, _ := cryptosuite.NewJSONWebKeySigner(aptJWK.ID, aptJWK.PrivateKeyJWK, cryptosuite.Authentication)
+	aptVerifier, _ := cryptosuite.NewJSONWebKeyVerifier(aptJWK.ID, aptJWK.PublicKeyJWK)
+
+	// Government Issuer
+	govtDidPrivateKey, govtDIDKey, _ := did.GenerateDIDKey(crypto.Ed25519)
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:47]:
```suggestion
	govtDIDPrivateKey, govtDIDKey, _ := did.GenerateDIDKey(crypto.Ed25519)
```

File: example/use_cases/apartment_application/apartment_application.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	govtJWK, _ := cryptosuite.JSONWebKey2020FromEd25519(govtDidPrivateKey.(ed25519.PrivateKey))
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:48]:
```suggestion
	govtJWK, _ := cryptosuite.JSONWebKey2020FromEd25519(govtDIDPrivateKey.(ed25519.PrivateKey))
```

File: example/use_cases/apartment_application/apartment_application.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	govtSigner, _ := cryptosuite.NewJSONWebKeySigner(govtJWK.ID, govtJWK.PrivateKeyJWK, cryptosuite.Authentication)
+
+	fmt.Print("\n\nStep 1: Create new DIDs for entities\n\n")
+	fmt.Printf("Tenant: %s\n", string(*holderDIDKey))
+	fmt.Printf("Apartment: %s\n", string(*aptDIDKey))
+	fmt.Printf("Government: %s\n", string(*govtDIDKey))
+
+	/**
+		 Step 2: Government issuer issues credentials to holder claiming age. The government issuer then signs the verifiable credentials to holder claiming age
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:57]: nit: `issuer issues a credential to the holder providing their age. The government...`

File: example/use_cases/apartment_application/apartment_application.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	vcBuilder := credential.NewVerifiableCredentialBuilder()
+
+	vcBuilder.SetIssuer(string(*knownIssuer))
+	vcBuilder.SetIssuanceDate(knownIssuanceDate)
+	vcBuilder.SetCredentialSubject(knownSubject)
+
+	vc, _ := vcBuilder.Build()
+	assert.NoError(nil, vc.IsValid())
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:74]: no assertions in non-test files 😄

File: example/use_cases/apartment_application/apartment_application.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+	signedVCBytes, _ := signing.SignVerifiableCredentialJWT(*govtSigner, *vc)
+
+	fmt.Print("\n\nStep 2: Government issues Verifiable Credential new for tenant verifying birthdate and signs\n\n")
+	if dat, err := util.PrettyJSON(vc); err == nil {
+		fmt.Printf("Verifiable Credential:%s\n", string(dat))
+	}
+
+	/**
+		Step 3: Create presentation definition from the apartment to the holder which goes into a presentation request.
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:84]: maybe language fix to make it clear that the physical apartment building is not making the request 😉

File: example/use_cases/apartment_application/apartment_application.go [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+				{
+					Path: []string{"$.credentialSubject.birthdate"},
+					ID:   "birthdate",
+				},
+			}},
+		},
+	})
+
+	presentationDefinition, _ := presentationDefinitionBuilder.Build()
+	assert.NoError(nil, presentationDefinition.IsValid())
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:107]: remove

File: example/use_cases/apartment_application/apartment_application.go [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	/**
+		Step 4: Tenant holder verifies the presentation request from the apt is valid and then constructs and signs a presentation submission
+	**/
+
+	verifiedPresentationDefinition, err := exchange.VerifyPresentationRequest(aptVerifier, exchange.JWTRequest, presentationRequestBytes)
+	assert.NoError(nil, verifiedPresentationDefinition.IsValid())
+
+	// TODO: Have the presentation claim's token format support signedVCBytes for the BuildPresentationSubmission function
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:123]: nit: for TODOs, generally advisable to link to an issue (or at least create one). I know I haven't been great about this either.

**Review by decentralgabe — CHANGES_REQUESTED** [review]
mostly small comments. looks great.

File: example/use_cases/apartment_application/apartment_application.go [hunk 9]
```diff
@@ -20,41 +20,47 @@ import (
...
-		 Step 1: Create new entities as DIDs. Govt Issuer, User Holder, and Apartment Verifier
+		 Step 1: Create new entities as DIDs. Govt Issuer, User Holder, and Apartment Verifier.
 	**/
 
 	// User Holder
-	holderDIDPrivateKey, holderDIDKey, _ := did.GenerateDIDKey(crypto.Ed25519)
+	holderDIDPrivateKey, holderDIDKey, err := did.GenerateDIDKey(crypto.Ed25519)
 	holderJWK, _ := cryptosuite.JSONWebKey2020FromEd25519(holderDIDPrivateKey.(ed25519.PrivateKey))
 	holderSigner, _ := cryptosuite.NewJSONWebKeySigner(holderJWK.ID, holderJWK.PrivateKeyJWK, cryptosuite.Authentication)
 	holderVerifier, _ := cryptosuite.NewJSONWebKeyVerifier(holderJWK.ID, holderJWK.PublicKeyJWK)
```
Comment (decentralgabe) [inline example/use_cases/apartment_application/apartment_application.go:38]: these should have err checks as well and all else below 😄

**Review by decentralgabe — APPROVED** [review]
excellent work!

**Comment by nitro-neal** [comment]
image

**Pull request merged**